ip/flac.c

```diff
@@ -207,19 +208,10 @@ static void metadata_cb(const Dec *dec, const FLAC__StreamMetadata *metadata, vo
 			const FLAC__StreamMetadata_StreamInfo *si = &metadata->data.stream_info;
 			int bits = 0;
 
-			switch (si->bits_per_sample) {
-			case 8:
-				bits = 8;
-				break;
-			case 12:
-			case 16:
-				bits = 16;
-				break;
-			case 20:
-			case 24:
-			case 32:
-				bits = 32;
-				break;
+			if (si->bits_per_sample >= 4 && si->bits_per_sample <= 32) {
+				bits = priv->bps = si->bits_per_sample;
+				if (bits % 8 > 0)
+				    bits = bits + (8 - bits % 8);
```

Comment:
`bits = 8 * ((bits + 7) / 8)` should do the trick without the conditional, right?

Reply:
Probably. I am not sure if it is better though. But if you insist, I can make the change.

Reply:
Please do, as it is a bit more readable by the virtue of being a well-known formula.

Reply:
Done.